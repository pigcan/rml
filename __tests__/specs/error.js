
const MLTransformer = require('../../src/Transformer');

describe('MLTransformer error', () => {
  it('report on attribute', (done) => {
    new MLTransformer([
      '<div prop="{{var}}" x="1">',
      '<div>',
      '1',
      '</div>',
      '</div>',
    ].join('\n')).transform((err, code) => {
      expect(code).toEqual(undefined);
      expect(err.message).toMatchSnapshot();
      done();
    });
  });

  it('report on attribute with text as child', (done) => {
    new MLTransformer([
      '<div prop="{{var}}" x="1">',
      '1',
      '</div>',
    ].join('\n')).transform((err, code) => {
      expect(code).toEqual(undefined);
      expect(err.startIndex).toEqual(0);
      expect(err.message).toMatchSnapshot();
      done();
    });
  });

  it('report on text node', (done) => {
    new MLTransformer([
      '<div prop="{{1}}" x="1">',
      '12{{var}}34',
      '<div>',
      '1',
      '</div>',
      '</div>',
    ].join('\n')).transform((err, code) => {
      expect(code).toEqual(undefined);
      expect(err.message).toMatchSnapshot();
      done();
    });
  });
});
