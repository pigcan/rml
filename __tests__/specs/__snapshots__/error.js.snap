// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MLTransformer error report on attribute 1`] = `"parse tag's attribute prop error: <div prop=\\"{{var}}\\" x=\\"1\\">"`;

exports[`MLTransformer error report on attribute with text as child 1`] = `"parse tag's attribute prop error: <div prop=\\"{{var}}\\" x=\\"1\\">"`;

exports[`MLTransformer error report on text node 1`] = `"parse text error: 12{{var}}34"`;
